okhttp-tests/src/test/java/com/squareup/okhttp/internal/ws/WebSocketReaderTest.java

```diff
@@ -123,6 +123,27 @@
     callback.assertTextMessage("Hello");
   }
 
+  @Test public void clientFramePayloadShort() throws IOException {
+    data.write(ByteString.decodeHex("817E000548656c6c6f")); // Hello
+    clientReader.processNextFrame();
+    callback.assertTextMessage("Hello");
+  }
+
+  @Test public void clientFramePayloadLong() throws IOException {
+    data.write(ByteString.decodeHex("817f000000000000000548656c6c6f")); // Hello
+    clientReader.processNextFrame();
+    callback.assertTextMessage("Hello");
+  }
+
+  @Test public void clientFramePayloadTooLongThrows() throws IOException {
+    data.write(ByteString.decodeHex("817f8000000000000000"));
+    try {
+      clientReader.processNextFrame();
+    } catch (IllegalStateException e) {
```

Comment:
Dunno. It's not technically a protocol problem.
On Mar 3, 2015 3:11 PM, "Jesse Wilson" <notifications@github.com> wrote:

> In
> okhttp-tests/src/test/java/com/squareup/okhttp/internal/ws/WebSocketReaderTest.java
> <https://github.com/square/okhttp/pull/1465#discussion_r25732371>:
>
> > +    data.write(ByteString.decodeHex("817E000548656c6c6f")); // Hello
> > +    clientReader.processNextFrame();
> > +    callback.assertTextMessage("Hello");
> > +  }
> > +
> > +  @Test public void clientFramePayloadLong() throws IOException {
> > +    data.write(ByteString.decodeHex("817f000000000000000548656c6c6f")); // Hello
> > +    clientReader.processNextFrame();
> > +    callback.assertTextMessage("Hello");
> > +  }
> > +
> > +  @Test public void clientFramePayloadTooLongThrows() throws IOException {
> > +    data.write(ByteString.decodeHex("817f8000000000000000"));
> > +    try {
> > +      clientReader.processNextFrame();
> > +    } catch (IllegalStateException e) {
>
> Is this the right exception? I would expect a ProtocolException
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/okhttp/pull/1465/files#r25732371>.
>